arm/Microsoft.DomainServices/AzureADDomainServices/deploy.bicep

```diff
@@ -0,0 +1,198 @@
+param secureldap string = 'Enabled'
+
+@description('Optional. Resource ID of the diagnostic storage account.')
+param diagnosticStorageAccountId string = ''
+
+@description('Optional. Resource ID of the diagnostic log analytics workspace.')
+param diagnosticWorkspaceId string = ''
+
+@description('Optional. Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.')
+param diagnosticEventHubAuthorizationRuleId string = ''
+
+@description('Optional. Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category.')
+param diagnosticEventHubName string = ''
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. Specifies the number of days that logs will be kept for; a value of 0 will retain data indefinitely.')
+@minValue(0)
+@maxValue(365)
+param diagnosticLogsRetentionInDays int = 365
+
+@allowed([
+  'CanNotDelete'
+  'NotSpecified'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = 'NotSpecified'
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'')
+param roleAssignments array = []
+
+@description('Optional. The name of logs that will be streamed.')
+@allowed([
+  'SystemSecurity'
+  'AccountManagement'
+  'LogonLogoff'
+  'ObjectAccess'
+  'PolicyChange'
+  'PrivilegeUse'
+  'DetailTracking'
+  'DirectoryServiceAccess'
+  'AccountLogon'
+])
+param logsToEnable array = [
+  'SystemSecurity'
+  'AccountManagement'
+  'LogonLogoff'
+  'ObjectAccess'
+  'PolicyChange'
+  'PrivilegeUse'
+  'DetailTracking'
+  'DirectoryServiceAccess'
+  'AccountLogon'
+]
+
+var diagnosticsLogs = [for log in logsToEnable: {
+  category: log
+  enabled: true
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+resource domainName_resource 'Microsoft.AAD/DomainServices@2021-05-01' = {
+  name: domainName
+  location: location
+  tags: tags
+  properties: {
+    domainName: domainName
+    domainConfigurationType: domainConfigurationType
+    filteredSync: filteredSync
+    notificationSettings: {
+      additionalRecipients: [
+        additionalRecipients
+      ]
+      notifyDcAdmins: notifyDcAdmins
+      notifyGlobalAdmins: notifyGlobalAdmins
+    }
+    ldapsSettings: {
+      externalAccess: ldapexternalaccess
+      ldaps: secureldap
+      pfxCertificate: pfxCertificate
+      pfxCertificatePassword: pfxCertificatePassword
+    }
+    replicaSets: [
```

Comment:
Note @mararam, here is another comment (which may be hidden by GitHub). I hope this comment uncovers it again.

Reply:
@MrMCake Yes, I'm a bit lost with this. I tried updating the parameter to support this, but I am seeing a compilation error for this. Could you please guide me to fix this?

Reply:
All you 'should' be required to do is to add a new parameter
```bicep
@description('<...>')
param replicaSets array
```

then, update the reference here to
```bicep
replicaSets: replicaSets
```

and update your parameter file to now provide this array as a parameter
```json
"replicaSets": {
  "value": [
    {
      "location": "..."
      "subnetId": "..."
    }
  ]
}
```

Reply:
@MrMCake thank you for the guidance. yes, I was almost there. I have updated the main.bicep and params file now. As part of this, I have removed the subnetId as the param(since this will be passed through replica sets now). Please review.